cpp/game_main.cpp

```diff
@@ -346,16 +346,27 @@ bool GameMain::on_input(SDL_Event *e) {
 			if (this->building_placement) {
 
 				// confirm building placement with left click
-				Command cmd(this->players[0], this->datamanager.get_producer_index(this->editor_current_building), mousepos_phys3);
-				cmd.set_ability(ability_type::build);
-				selection.all_invoke(cmd);
+				// first create foundation using the producer
+				auto player = this->selection.owner();
```

Comment:
I really don't like this exaggerated use of `auto`; it takes away the code's self-documenting properties. For example, to check whether this line doesn't actually call a copy constructor, I have to to open `game_main.h` to find out what `this->selection` is, then open `unit/selection.h` to find out what `UnitSelection.owner()` returns...